bot/storage/storage.py

```diff
@@ -0,0 +1,61 @@
+import datetime
+import logging
+
+class Storage(object):
+    """ Virtual Base Class for Storage Implementations.
+        
+        Minimal implementation requires overriding the following:
+            _getRecord(..)
+            _storeRecord(..)
+            initialize(..) (if nec)
+            
+        Sharing _getEmptyContacts(..) ensures that all storage implementations
+        share basic properties.
+    """
+    
+    def _getRecord(self, userid):
+        ''' Takes userid and returns record if it exists, otherwise and 
+        blank contact record (see _getEmptyContact(..)). '''
+        
+        raise Exception("_getRecord must be implemented through child class")
+
+    def _storeRecord(self, record):
+        ''' Takes contact record and stores it. See _getEmptyContact for record format.'''
+        raise Exception("_storeRecord must be implemented through child class")
+
+    def _getEmptyContact(self, userid):
+        return {
+            'id': userid,
+            'contact': "No contact information stored",
+            'access':[],
+            'context':{} # this is solely to allow for "grepping" in case of
+            # emergency
+        }
+        
+    def initialize(self):
+        pass # does not need to be implemented
```

Comment:
Two-phase initialization is not great, why can't the constructor handle this?

Reply:
Initialization is specific to the concrete child storage class...

Reply:
So is the constructor (`__init__`)